spec/booking_system_spec.rb

```diff
@@ -0,0 +1,299 @@
+      room2_before = @booking.rooms[1].reservations.count
+      no_overlap_before
+      no_overlap_after
+      room1_after = @booking.rooms[0].reservations.count
+      room2_after = @booking.rooms[1].reservations.count
+
+      expect(room1_after).must_equal room1_before + 2
+      expect(room2_after).must_equal room2_before
+    end
+
+    it "selects the same room when requested date ends on check_in date" do
+      room1_before = @booking.rooms[0].reservations.count
+      room2_before = @booking.rooms[1].reservations.count
+      end_on_checkin
+      room1_after = @booking.rooms[0].reservations.count
+      room2_after = @booking.rooms[1].reservations.count
+
+      expect(room1_after).must_equal room1_before + 1
+      expect(room2_after).must_equal room2_before
+    end
+
+    it "selects the same room when requested date starts on check_out date" do
+      room1_before = @booking.rooms[0].reservations.count
+      room2_before = @booking.rooms[1].reservations.count
+      end_on_checkin
+      room1_after = @booking.rooms[0].reservations.count
+      room2_after = @booking.rooms[1].reservations.count
+
+      expect(room1_after).must_equal room1_before + 1
+      expect(room2_after).must_equal room2_before
+    end
+
+    it 'assigns a reservation id to the new reservation consecutively' do
+      no_overlap_before
+      no_overlap_after
+      expect(no_overlap_after.id).must_equal no_overlap_before.id + 1
+    end
+
+    it 'raises an exception when there are no rooms available for specified date' do
+      check_in = "Oct 4 2018"
+      check_out = "Oct 7 2018"
+
+      19.times do |i|
+        @booking.reserve_room(check_in, check_out)
+      end
+
+      expect {
+        @booking.reserve_room(check_in, check_out)
+      }.must_raise ArgumentError
+    end
+
+    it 'selects another room when requested date overlaps with block' do
+      room1_before = @booking.rooms[0].reservations.count
+      room2_before = @booking.rooms[1].reservations.count
+      room3_before = @booking.rooms[2].reservations.count
+      room4_before = @booking.rooms[3].reservations.count
+      room5_before = @booking.rooms[4].reservations.count
+      room6_before = @booking.rooms[5].reservations.count
+      overlap_block
+      room1_after = @booking.rooms[0].reservations.count
+      room2_after = @booking.rooms[1].reservations.count
+      room3_after = @booking.rooms[2].reservations.count
+      room4_after = @booking.rooms[3].reservations.count
+      room5_after = @booking.rooms[4].reservations.count
+      room6_after = @booking.rooms[5].reservations.count
+
+      expect(room1_after).must_equal room1_before
+      expect(room2_after).must_equal room2_before
+      expect(room3_after).must_equal room3_before
+      expect(room4_after).must_equal room4_before
+      expect(room5_after).must_equal room5_before
+      expect(room6_after).must_equal room6_before + 1
+    end
+  end
+
+  describe 'date_range' do
```

Comment:
Since the user would never be calling this method (it's really a method used internally to the booking system, I would make the method `private` and not test it directly.